src/main/java/edu/cornell/kfs/krad/service/impl/ClamAVAntiVirusServiceImpl.java

```diff
@@ -0,0 +1,217 @@
+            throws IOException {
+        sendSimpleCommandToClamAV(command, socketOutput);
+        socketOutput.flush();
+        return readClamAVResponse(socketReader);
+    }
+
+    private void sendSimpleCommandToClamAV(String command, DataOutputStream socketOutput) throws IOException {
+        if (StringUtils.isBlank(command)) {
+            throw new IllegalArgumentException("command cannot be blank");
+        }
+        String delimitedCommand = ClamAVDelimiters.Z_PREFIX + command + ClamAVDelimiters.NULL_SUFFIX;
+        byte[] delimitedCommandAsBytes = delimitedCommand.getBytes(StandardCharsets.UTF_8);
+        socketOutput.write(delimitedCommandAsBytes);
+    }
+
+    private String readClamAVResponse(Reader socketReader) throws IOException {
+        char[] responseChunk = new char[DEFAULT_CHUNK_SIZE];
+        StringBuilder fullResponse = new StringBuilder(DEFAULT_CHUNK_SIZE);
+        int responseTerminatorIndex = -1;
+        int chunkLength;
+        
+        do {
+            chunkLength = socketReader.read(responseChunk);
+            if (chunkLength > 0) {
+                fullResponse.append(responseChunk, 0, chunkLength);
+                responseTerminatorIndex = fullResponse.indexOf(
+                        ClamAVDelimiters.NULL_SUFFIX, fullResponse.length() - chunkLength);
+            }
+        } while (responseTerminatorIndex == -1 && chunkLength > 0);
+        
+        if (responseTerminatorIndex == -1) {
+            LOG.error("readClamAVResponse, End of stream was reached before reading full ClamAV response!  " +
+                    "No content will be returned.  " + fullResponse.length() +
+                    " characters were received prior to abrupt end.");
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("readClamAVResponse, Response text received prior to stream end: " + fullResponse);
+            }
+            return KFSConstants.EMPTY_STRING;
+        } else if (responseTerminatorIndex < fullResponse.length() - 1) {
+            int extraContentLength = fullResponse.length() - responseTerminatorIndex - 1;
+            LOG.warn("readClamAVResponse, ClamAV sent at least an extra " + extraContentLength +
+                    " characters of response data beyond the delimiter!  The extra characters will be ignored.");
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("readClamAVResponse, Response text received beyond end-of-content delimiter: " +
+                        fullResponse.substring(responseTerminatorIndex + 1));
+            }
+        }
+        
+        String responsePrecedingTerminator = fullResponse.substring(0, responseTerminatorIndex);
+        return responsePrecedingTerminator;
+    }
+
+    @Override
+    public ClamAVScanResult scan(byte[] fileContents) throws IOException {
+        try (
+            InputStream preLoadedFileStream = new ByteArrayInputStream(fileContents);
+        ) {
+            return scan(preLoadedFileStream);
+        }
```

Comment:
no catch here?

Reply:
The scan() method that accepts a byte array already had "throws IOException" in its method signature, so I thought that it would not be necessary to perform a catch here. Or, if it's just unclear as to why this section doesn't have a "catch" or "finally" block at all, it's because I had set it up as a try-with-resources block, which can allow for omitting an explicit catch/finally block afterwards.

Reply:
It was unclear why there was not catch or finally block I don't often see a standalone try block

Reply:
I'll add a "catch" block here, to avoid confusion and to also provide a spot for logging the exception before re-throwing it.